crypto/encrypt/encrypt.go

```diff
@@ -38,21 +51,25 @@ func Bytes(bytes []byte) ([]byte, []byte, error) {
 	// gcm or Galois/Counter Mode, is a mode of operation
 	// for symmetric key cryptographic block ciphers
 	// - https://en.wikipedia.org/wiki/Galois/Counter_Mode
-	gcm, err := cipher.NewGCM(block)
+	aesGCM, err := cipher.NewGCM(block)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	// Create a new byte array the size of the nonce,
 	// populate with a cryptographically secure random sequence
-	nonce := make([]byte, gcm.NonceSize())
-	_, err = rand.Read(nonce)
-	if err != nil {
-		return nil, nil, err
+	if params.Nonce == nil {
```

Comment:
Question, do we need this? The Nonce/IV would be returned anyway, correct? So we could first try to compute the secret in the test, then use the returned nonce and key to try and `Open` the `aesGCM` and assert the text matches? I think that might be a simpler approach/assertion and not require much change in this original file. Thoughts @jeremyletang, @rickymarcon?

Reply:
well if the nonce is passed as a parameter only for the test, then yea eventually. Although depending of the API for your package you may also wnat to be able to specify your own nonce when call the functions I would suppose.

Reply:
Sorry, I mean - if the nonce/iv is not returned, we can add it which might make the whole thing a bit simpler to test.

Reply:
Sorry, I mean - if the nonce/iv is not returned, we can add it which might make the whole thing a bit simpler to test.

Reply:
And just wondering - would it not be better to NOT _add_ parameters just for testing, but rather _return_ more attributes - which will be helpful for testing? Since we don't support adding own nonce now I'd prefer not adding it as it introduces more if statements and therefore branches to test etc.

Reply:
A&A. Updated. WDYT.